src/main/java/com/sintern/service/LogoServiceImpl.java

```diff
@@ -0,0 +1,52 @@
+package com.sintern.service;
+
+import com.sintern.domain.entity.Company;
+import com.sintern.domain.entity.CompanyLogo;
+import com.sintern.exception.LogoCompanyException;
+import com.sintern.repository.CompanyRepository;
+import com.sintern.repository.LogoRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+import org.springframework.web.multipart.MultipartFile;
+
+import javax.transaction.Transactional;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+public class LogoServiceImpl implements LogoService {
+
+    private final LogoRepository logoRepository;
+    private final CompanyRepository companyRepository;
+
+    @Autowired
+    public LogoServiceImpl(LogoRepository logoRepository, CompanyRepository companyRepository) {
+        this.logoRepository = logoRepository;
+        this.companyRepository = companyRepository;
+    }
+
+    @Override
+    @Transactional
```

Comment:
Just curious. Why is `@Transactional` needed here?

Reply:
in short: Optional <Company Logo> is returned